lang/src/main/antlr4/org/kaazing/k3po/lang/parser/v2/Robot.g4

```diff
@@ -41,7 +41,11 @@ acceptableNode
     ;
 
 connectNode
-    : k=ConnectKeyword (AwaitKeyword barrier=Name ConnectKeyword)? connectURI=location (OptionKeyword TransportKeyword value=location)? streamableNode+
+    : k=ConnectKeyword (AwaitKeyword barrier=Name ConnectKeyword)? connectURI=location
+                       (OptionKeyword TransportKeyword value=location)?
+                       (OptionKeyword SizeKeyword DecimalLiteral)?
+                       (OptionKeyword ModeKeyword ModeValue)?
+        streamableNode+
     ;
```

Comment:
Yes, that was the intention. "r" is for read only mode. We open the file and this parameter corresponds to mode parameter in RandomAccessFile(File, String mode). I think this would also protect against accidentally modifying the file. Otherwise, we would be opening the file in read-write mode always even if the script just wants to do read operations.

Regarding whether we want to have an explicit option for creation, I thought "rw" mode can be used just like RandomAccessFile.

Other thing is whether we want the script to be more expressive instead of "r", "rw". We can have something like "read-only", "read-write" etc. But I think mode is very common.